src/Test/Utilities/Portable/Platform/CoreClr/TestExecutionLoadContext.cs

```diff
@@ -37,57 +36,22 @@ public TestExecutionLoadContext(IList<ModuleData> dependencies)
 
         protected override Assembly Load(AssemblyName assemblyName)
         {
-            lock (s_platformAssemblyPaths)
+            var comparer = StringComparer.OrdinalIgnoreCase;
+            var comparison = StringComparison.OrdinalIgnoreCase;
+            if (assemblyName.Name.StartsWith("System.", comparison) ||
+                assemblyName.Name.StartsWith("Microsoft.", comparison) ||
+                comparer.Equals(assemblyName.Name, "mscorlib") ||
+                comparer.Equals(assemblyName.Name, "netstandard"))
             {
-                Assembly assembly;
-                if (s_platformAssemblyPaths.TryGetValue(assemblyName.Name, out var assemblyPath))
-                {
-                    assembly = LoadPlatformAssembly(assemblyName, assemblyPath);
-                    if (assembly != null)
-                    {
-                        return assembly;
-                    }
-                }
-
-                if (_dependencies.TryGetValue(assemblyName.FullName, out var moduleData))
-                {
-                    return LoadImageAsAssembly(moduleData.Image);
-                }
-
                 return null;
             }
-        }
 
-        private Assembly LoadPlatformAssembly(AssemblyName assemblyName, string assemblyPath)
-        {
-            lock (s_loadedPlatformAssemblies)
+            if (_dependencies.TryGetValue(assemblyName.FullName, out var moduleData))
             {
-                if (s_loadedPlatformAssemblies.TryGetValue(assemblyPath, out var assembly))
-                {
-                    return assembly;
-                }
-                else
-                {
-                    try
-                    {
-                        assembly = Default.LoadFromAssemblyName(assemblyName);
-                    }
-                    catch (FileNotFoundException)
-                    {
-                        // The assembly wasn't in the TPA list we can try to load
-                        // it from the dependencies list of the assembly. However,
-                        // if the assembly is mscorlib we won't be able to reload it,
-                        // no matter what
-                        if (assemblyName.Name == "mscorlib")
-                        {
-                            throw;
-                        }
-                        assembly = null;
-                    }
-
-                    return assembly;
-                }
+                return LoadImageAsAssembly(moduleData.Image);
```

Comment:
No idea if this is thread-safe.

Reply:
What threading operation are you concerned about here? The `moduleData` value is immutable so it's fine. The eventual call to `LoadFromStream` appears to be fine for use in this manner. The runtime test bed uses this exact pattern https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/System.Runtime.Loader/tests/ResourceAssemblyLoadContext.cs

Reply:
I was specifically worried about re-entrancy. Sounds good.